portalnet/src/storage.rs

```diff
@@ -67,6 +67,13 @@ pub enum ContentStoreError {
     ContentKey(#[from] ContentKeyError),
 }
 
+#[derive(Debug, PartialEq)]
+pub enum ShouldWeStoreContent {
```

Comment:
The thing that stands out to me about the name is that it is conflating two things:
1. current status of the content
2. the proposed action as a result of that content status

There might be a variety of reasons that callers want to know the current status of the content, besides just "should we store it". Notice the function that returns it isn't called `should_store()`, because there might be other reasons you want to know this info. I opened a PR with a recommended name change: #983 

---

All that said, this PR clearly is a great improvement from the status quo.